### PR TITLE
Do not crash if RichText has recognizers without handlers

### LINKED ISSUE #69788: CastError in SemanticsConfiguration.onTap
While reproducing https://github.com/flutter/flutter/issues/69787 I also found the following (unrelated) crash:

```dart
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Directionality(
      textDirection: TextDirection.ltr,
      child: Center(
        child: Container(
          color: Colors.green,
          child: RichText(
            text: TextSpan(
              text: 'Start\n',
              recognizer: TapGestureRecognizer(),
            ),
          ),
        ),
      ),
    );
  }
}
```

throws (with semantics turned on):

```
I/flutter (23037): ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (23037): The following _CastError was thrown during a scheduler callback:
I/flutter (23037): Null check operator used on a null value
I/flutter (23037): 
I/flutter (23037): When the exception was thrown, this was the stack:
I/flutter (23037): #0      SemanticsConfiguration.onTap= (package:flutter/src/semantics/semantics.dart:2864:54)
I/flutter (23037): #1      RenderParagraph.assembleSemanticsNode (package:flutter/src/rendering/paragraph.dart:939:27)
I/flutter (23037): #2      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3717:13)
I/flutter (23037): #3      _RootSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3587:16)
I/flutter (23037): #4      RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2631:25)
I/flutter (23037): #5      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1081:16)
I/flutter (23037): #6      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:459:21)
I/flutter (23037): #7      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:895:13)
I/flutter (23037): #8      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:320:5)
I/flutter (23037): #9      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
I/flutter (23037): #10     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
I/flutter (23037): #11     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:864:7)
I/flutter (23037): (elided 11 frames from class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
I/flutter (23037): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/69788.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -935,13 +935,19 @@ class RenderParagraph extends RenderBox
         final GestureRecognizer? recognizer = info.recognizer;
         if (recognizer != null) {
           if (recognizer is TapGestureRecognizer) {
-            configuration.onTap = recognizer.onTap;
```
Comment (chunhtai) [inline packages/flutter/lib/src/rendering/paragraph.dart:938]: not related to this PR. It feels weird to me configuration.onTap is a nullable value but will crash if you really give it null.

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/paragraph.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/rendering/paragraph.dart:938]: Yeah, the problem is that the `onTap` getter must be nullable, but the setter is not really and dart doesn't allow specifying that...

**Review by chunhtai — APPROVED** [review]

File: packages/flutter/lib/src/rendering/paragraph.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/rendering/paragraph.dart:938]: Do we have use case where we want to reset configuration.onTap back to null?

**Pull request merged**